envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/clusters/EnvoyClustersFactory.kt

```diff
@@ -103,7 +104,56 @@ class EnvoyClustersFactory(
     }
 
     fun getClustersForGroup(group: Group, globalSnapshot: GlobalSnapshot): List<Cluster> =
-        getEdsClustersForGroup(group, globalSnapshot) + getStrictDnsClustersForGroup(group)
+        getEdsClustersForGroup(group, globalSnapshot) + getStrictDnsClustersForGroup(group) + getClustersForJWT()
+
+    private fun getClustersForJWT() = properties.jwt.providers.map(this::clusterForOAuthProvider)
```

Comment:
It looks like this cluster is build only when properties change. That requires application restart. Maybe lets build this cluster at the beginning so we won't have to build each time.